ec/src/hashing/curve_maps/wb/mod.rs

```diff
@@ -0,0 +1,90 @@
+use core::marker::PhantomData;
+
+use ark_ff::vec::Vec;
+use ark_poly::{Polynomial, UVPolynomial, univariate::{DensePolynomial}};
+use crate::models::SWModelParameters;
+use crate::ModelParameters;
+
+use crate::hashing::map_to_curve_hasher::MapToCurve;
+use crate::hashing::HashToCurveError;
+use crate::{AffineCurve};
+use crate::models::short_weierstrass_jacobian::GroupAffine;
+
+use super::swu::{SWUParams, SWUMap};
+
+/// Implementation for the WB hash to curve for the curves of Weierstrass form of y^2 = x^3 + a*x + b where b != 0 but a can be zero like BLS-381 curve. From [WB2019]
+///
+/// - [WB19] Wahby, R. S., & Boneh, D. (2019). Fast and simple constant-time
+///   hashing to the bls12-381 elliptic curve. IACR Transactions on
+///   Cryptographic Hardware and Embedded Systems, nil(nil), 154–179.
+///   http://dx.doi.org/10.46586/tches.v2019.i4.154-179
+///
+///
+pub trait WBParams : SWModelParameters + Sized
+{
+    //The isogenouscurve should be defined over the same basefield but it can have different scalar field
+    //type IsogenousCurveScalarField :
+    type IsogenousCurve : SWUParams::<BaseField = <Self as ModelParameters>::BaseField>;
+
+    const PHI_X_NOM: &'static [<Self::IsogenousCurve as ModelParameters>::BaseField];
+    const PHI_X_DEN: &'static [<Self::IsogenousCurve as ModelParameters>::BaseField];
+    
+    const PHI_Y_NOM: &'static [<Self::IsogenousCurve as ModelParameters>::BaseField];
+    const PHI_Y_DEN: &'static [<Self::IsogenousCurve as ModelParameters>::BaseField];
```

Comment:
I'm not sure what order "small" refers to. Currently the largest isogeny we have encountered is of degree 23. It is for the case of  BLS12-377 G2. This means PHI_Y could have of up to degree 46 numerator and denominator with each coefficient of 377bit size: https://github.com/w3f/arkworks-curves/blob/a22aa14c2306ee23ff9f36ed639a9a969daa157b/bls12_377/src/curves/g2.rs#L242.  However, one might consider even larger degree isogenies.

However because this evaluation carried out twice each time we hash something to the curve, any slight optimization makes a difference for protocols which need hashing lots of values to the curve. In that case, the benefit of using  `ark_poly` outweights the extra dependency concern, if it provides even a small optimization over naïve evaluation.

Reply:
I see, let's just use `ark-poly` then. I just wanted to avoid any circular dependency issues, but it seems that there aren't any right now.